networkx/algorithms/d_separation.py

```diff
@@ -140,3 +160,190 @@ def d_separated(G, x, y, z):
         return False
     else:
         return True
+
+
+@not_implemented_for("undirected")
+def minimal_d_separator(G, u, v):
+    """Compute a minimal d-separating set between X and Y.
+
+    A d-separating set in a DAG is a set that when conditioned on
+    blocks all paths between the two sets of variables. This function
+    constructs a set that is "minimal", meaning it is the smallest
+    d-separating set between the two variables. This is not necessarily
+    unique. For more details, see Notes.
+
+    Parameters
+    ----------
+    G : graph
+        A networkx DAG.
+    u : node
+        A node in the graph, G.
+    v : node
+        A node in the graph, G.
+
+    Raises
+    ------
+    NetworkXError
+        The *d-separation* test is commonly used with directed
+        graphical models which are acyclic.  Accordingly, the algorithm
+        raises a :exc:`NetworkXError` if the input graph is not a DAG.
+
+    NodeNotFound
+        If any of the input nodes are not found in the graph,
+        a :exc:`NodeNotFound` exception is raised.
+
+    References
+    ----------
+    .. [1] Tian, J., & Paz, A. (1998). Finding Minimal D-separators.
+
+    Notes
+    -----
+    This function only finds ``a`` minimal d-separator. It does not guarantee
+    uniqueness, since in a DAG there may be more than one minimal d-separator
+    between two nodes. Moreover, this only checks for minimal separators
+    between two nodes, not two sets. Finding minimal d-separators between
+    two sets of nodes is not supported.
+
+    Uses the algorithm presented in [1]_. The complexity of the algorithm
+    is :math:`O(|E_{An}^m|)`, where :math:`E_{An}^m` stands for the
+    number of edges in the moralized graph of the sub-graph consisting
+    of only the ancestors of X and Y. For full details, see [1]_.
+
+    The algorithm works by constructing the moral graph consisting of just
+    the ancestors of ``u`` and ``v``. Then it constructs a candidate for
+    a separating set from the predecessors of ``u`` and ``v``. Starting
+    from this set, ``Z'``, BFS is ran starting from ``u`` and marking nodes
+    it finds in ``Z'``. The resulting nodes marked, ``Z''``, are returned.
+    Then BFS is ran again starting from ``v`` and marking nodes if they are
+    present in ``Z''``. The returned set of marked nodes is a minimal
+    d-separating set.
+
+    https://en.wikipedia.org/wiki/Bayesian_network#d-separation
+    """
+    if not nx.is_directed_acyclic_graph(G):
+        raise nx.NetworkXError("graph should be directed acyclic")
+
+    union_xy = set(u).union(set(v))
```

Comment:
Would this be more clear?
`union_xy = {u, v}`

And below: `D_anc_xy = D_anc_xy.union(u, v))` could be `D_anc_xy.update((u, v))`

Reply:
Done